yokozach/The-Mechanical-Forest
# Save File Creation and Game Config File

**Changes:**
- Broke down `SaveMenuController` prefab into more prefabs including `SaveFilecontainer` and `SaveMenuPanel`.
- Created `ConfigBaseData` `ScriptableObject` which is used for internal game config settings such as file/folder paths for persistent data that should not be modifiable by the player.
- Created `DefaultConfig` settings, which sets out filepaths and filenames for our save data.
- Created `ConfigMonoSystem` and `IConfigMonoSystem`, which are used to retrieve information from the active configuration file via dependency injection through the `GameManager`.
- Temporarily made many fields in the `PartyMemberController`, `PartyMemberStatsHandler`, and `PartyMemberResourceHandler` public. **This will need to be changed later** to protect these fields via `getters` and `setters`.
- Created a `PartyMemberConverter` `static class` in `Runtime.Converters` to convert between `JSON serializable` `PartyMember` and `PartyMemberController`. This helps when creating a `SaveFile` after retrieving the `PartyMemberControllers` from the `PartyMonoSystem`. Separating this from the `JSON serializable` class also keeps the class cleaner.
- Drew the rest of the owl to make `SaveFileContainer` query `SaveMonoSystem` for the currently existing `SaveFile`s and use this information to fill out each `SaveSlot` with their info.
- Also modified `SaveSlot` to create a new party through the `PartyMonoSystem` and convert the resultant `List` of `PartyMemberController`s to `PartyMembers` and place these into a `SaveFile` which then get's passed to the `SaveMonoSystem` on `InitializeNewFile`.
- Modified `SaveMonoSystem` to `SaveFileAsync` on `InitializeNewFile`.
- Added a field to our `SaveFile` to include the player designated `saveFileName`.

**Testing:**

Because we cannot yet load save files, testing is limited to the creation of new save files. On runtime, do the following:

1. Click the "New Game Button".
2. Attempt to create a "New Game"
3. Navigate to the persistent data path associated with this game and check that the file path for save files was created and that you successfully generated a new save file.

**For More Info See:**
- closes #94 
- closes #113
